ros/src/computing/planning/motion/packages/astar_planner/nodes/velocity_set/velocity_set.cpp

```diff
@@ -144,11 +144,32 @@ void displayDetectionRange(const autoware_msgs::lane& lane, const CrossWalk& cro
   waypoint_marker_decelerate.color.b = 0.0;
   waypoint_marker_decelerate.frame_locked = true;
 
+  bool found_obstacle = false, found_stopline = false;
   for (auto& info : obstacle_infos)
   {
+    // use nearest obstacle and stopline
+    if ((info.type == EObstacleType::ON_WAYPOINTS ||
+         info.type == EObstacleType::ON_CROSSWALK))
+    {
+      if (!found_obstacle) found_obstacle = true;
+      else continue;
+    }
+    if (info.type == EObstacleType::STOPLINE)
+    {
+      if (!found_stopline) found_stopline = true;
+      else continue;
+    }
+
```

Comment:
@aohsato why is this needed?

Reply:
for vizualizing only nearest stop line and obstacle. you saw doubled geo fence before this.